runtime-light/coroutine/awaitable.h

```diff
@@ -199,3 +200,41 @@ class wait_fork_t {
     }
   }
 };
+
+// ================================================================================================
+
+class wait_queue_next_t {
+  int64_t queue_id;
+  wait_for_timer_t timer_awaiter;
+
+public:
+  explicit wait_queue_next_t(int64_t queue_id_, std::chrono::nanoseconds timeout_) noexcept
+    : queue_id(queue_id_)
+    , timer_awaiter(timeout_) {}
+
+  bool await_ready() const noexcept {
+    if (auto queue = WaitQueueContext::get().get_queue(queue_id); queue.has_value()) {
+      return queue.val()->empty() || queue.val()->has_ready_fork();
+    }
+    return true;
+  }
+
+  void await_suspend(std::coroutine_handle<> coro) noexcept {
```

Comment:
There is something wrong with the design: `wait_queue_next_t` always waits for timeout before trying to check fork readiness.  So, for example, `wait_queue_next($id, 1sec)` is going to wait 1 second anyway, even if next fork is ready in 10ms.

Reply:
remove class `wait_queue_next_t`